src/components/organisms/GitCloneModal/GitCloneModal.tsx

```diff
@@ -55,27 +55,25 @@ const GitCloneModal: React.FC = () => {
         fs.mkdirSync(localPath, {recursive: true});
       }
 
-      promiseFromIpcRenderer('git.cloneGitRepo', 'git.cloneGitRepo.result', {
-        localPath: localGitPath,
-        repoPath: repoURL,
-      }).then(result => {
+      try {
+        await cloneGitRepo({localPath: localGitPath, repoPath: repoURL});
+        setLoading(false);
+        dispatch(closeGitCloneModal());
+        dispatch(setCreateProject({rootFolder: `${localPath}${sep}${repoName.replace('.git', '')}`}));
+      } catch (error: any) {
         setLoading(false);
         dispatch(closeGitCloneModal());
 
-        if (result.error) {
-          Modal.warning({
-            title: 'Clone failed!',
-            content: <div>{result.error}</div>,
-            zIndex: 100000,
-          });
-
-          if (doesPathExist(localGitPath)) {
-            fs.rmdirSync(localGitPath, {recursive: true});
-          }
-        } else {
-          dispatch(setCreateProject({rootFolder: `${localPath}${sep}${repoName.replace('.git', '')}`}));
+        Modal.warning({
+          title: 'Clone failed!',
+          content: <div>{error.message}</div>,
+          zIndex: 100000,
+        });
+
+        if (doesPathExist(localGitPath)) {
+          fs.rmdirSync(localGitPath, {recursive: true});
```

Comment:
do we have to do this in-sync? I'm guessing we could use the `async` version of this method